### PR TITLE
docs: fixes for release notes for 1.20

### PR DESCRIPTION
**Pull request by alyssawilk**
cleaning up release notes

Risk Level: n/a
Testing: n/a
Docs Changes: yes
Release Notes: no

### CODE REVIEW

File: VERSION
Comment (phlax) [inline VERSION:1]: i thought these changes happened on the release branch, rather than `main` - but not sure how its done

Comment (alyssawilk) [reply VERSION:1]: per GOVERNANCE.md it's supposed to be with the release PR
the one sketchy bit is me setting the dates to the "wrong day" but given the list we have I'd be shocked if we cut the release today and I'm being optimistic about aiming for tomorrow :-P

**Comment by phlax** [comment]
docs are rendered here https://storage.googleapis.com/envoy-pr/18403/docs/index.html

**Comment by phlax** [comment]
in "NEW FEATURES" there is a bad link for `envoy_v3_api_field_config.overload.v3.OverloadManager.buffer_factory_config` i think its missing a preceding space

**Comment by alyssawilk** [comment]
Hey @envoyproxy/envoy-maintainers  1.20.0 release notes here:
with one known reflink issue @phlax caught above.  PTAL and suggest other improvements.

https://storage.googleapis.com/envoy-pr/8e2d393/docs/version_history/current.html

**Comment by alyssawilk** [comment]
yeah I wonder if we should format check rst links better - that's two separate bugs in this release and I swear that's half the updates we do nowadays.  either way, fix coming soon!

**Comment by phlax** [comment]
> I wonder if we should format check rst links bette

i have a PR that does this, but i havent pushed it because i wasnt entirely happy with any of the rst linters, and didnt decide on a way forward

i will tho (and then hide when everyone moans that they have to wait for docs to be built before the rst formatting issues surface 8/)

**Comment by alyssawilk** [comment]
Hm, we could add checks for the 2 failure modes we've seen to the existing rst checks, so at least we don't need to run full docs build for release notes checks.  WDYT?

**Comment by phlax** [comment]
i think its probs a good idea - ie very shallow rst lint check in formatting checks, and then more thorough check once the rst has been compiled from protos etc

i think the bigger issue is that its not just this file, and running *any* linter against the current docs creates a humongous list of issues

**Comment by phlax** [comment]
some other thoughts on this are

- i dont want to re/implement our own linter with a ton of regexes, so im ambivalent about adding too many more checks, esp for only one file
- some checks simply cant be picked up by linters, as they cant know in advance if the rst is valid - eg links to anchors etc

**Comment by davinci26** [comment]
LGTM overall, modulo the broken links.

**Review by phlax — DISMISSED** [review]
lgtm

**Review by phlax — APPROVED** [review]
lgtm

File: RELEASES.md
Comment (suniltheta) [inline RELEASES.md:76]: Curious to know why this is 1.20.1 and not 1.21.0 ?

Comment (phlax) [reply RELEASES.md:76]: good spot @suniltheta - do you want to PR to amend ?

**Pull request merged**
